server/planning/search/planning_search.py

```diff
@@ -118,19 +95,56 @@ def _format_docs(self, docs):
                 if (doc["dates"].get("recurring_rule") or {}).get("until"):
                     doc["dates"]["recurring_rule"]["until"] = parse_date(doc["dates"]["recurring_rule"]["until"])
 
-    def _get_projected_fields(self, req):
-        """Get elastic projected fields."""
-        app = get_current_app()
-        if app.data.elastic.should_project(req):
-            return app.data.elastic.get_projected_fields(req)
+    def get_indexes_for_search(self, repos: list[str]) -> list[str]:
+        indexes = []
+        if "events" in repos:
+            indexes += EventResourceModel.get_service().elastic.config.index
+        if "planning" in repos:
+            indexes += PlanningResourceModel.get_service().elastic.config.index
+        return indexes
 
-    def _get_index(self, repos):
-        """Get index id for all repos."""
-        app = get_current_app()
-        indexes = {app.data.elastic.index}
-        for repo in repos:
-            indexes.add(app.config["ELASTICSEARCH_INDEXES"].get(repo, app.data.elastic.index))
-        return ",".join(indexes)
+    def get_projection(self, req) -> list[str] | None:
+        fields = self._get_projected_fields(req)
+        projection = None if not fields else fields.split(",")
+
+        if projection and "type" not in projection:
+            # Make sure `type` is always included in the projection
+            projection.append("type")
+
+        return projection
+
+    async def get_async(self, req, lookup):
+        """Run the query against events and planning indexes"""
+        query = self._get_query(req)
+        types = self._get_types(req)
+        indexes = self.get_indexes_for_search(types)
+        projection = self.get_projection(req)
+
+        elastic = EventResourceModel.get_service().elastic
+        docs = await elastic.search(query, indexes, projection)
```

Comment:
I have been working on something similar on the content authoring side (upgrading global content search to async), and there may need to be a couple more fixes here.

```python
from eve_elastic.elastic import fix_query

async def get_async(self, req, lookup):
    ...
    hits = await elastic.search(fix_query(query), indexes, projection)
    docs = self.elastic._parse_hits(hits, types[0])
```

You'll see 2 changes here:
* The use of `fix_query` from the `eve_elastic` library (as some front-end code still used Elastic version 1 and/or  2 queries  :face_exhaling: 
* Using the `_parse_hits` from the `eve_elastic` library to process the response

I found these 2 extra steps were needed so the response is in the correct format etc